src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToHtml5DateTimeLocalTransformer.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * @author Franz Wilding <franz.wilding@me.com>
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ */
+class DateTimeToHtml5DateTimeLocalTransformer extends BaseDateTimeTransformer
+{
+    /**
+     * Transforms a normalized date into a localized date without trailing timezone.
+     *
+     * According to the HTML standard, the input string of a datetime-local
+     * input is a RFC3339 date followed by 'T', followed by a RFC3339 time.
+     * http://w3c.github.io/html-reference/datatypes.html#form.data.datetime-local
+     *
+     * @param \DateTime|\DateTimeInterface $dateTime A DateTime object
+     *
+     * @return string The formatted date
+     *
+     * @throws TransformationFailedException If the given value is not an
+     *                                       instance of \DateTime or \DateTimeInterface
+     */
+    public function transform($dateTime)
+    {
+        if (null === $dateTime) {
+            return '';
+        }
+
+        if (!$dateTime instanceof \DateTime && !$dateTime instanceof \DateTimeInterface) {
+            throw new TransformationFailedException('Expected a \DateTime or \DateTimeInterface.');
+        }
+
+        if ($this->inputTimezone !== $this->outputTimezone) {
+            if (!$dateTime instanceof \DateTimeImmutable) {
+                $dateTime = clone $dateTime;
+            }
+
+            $dateTime = $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
+        }
+
+        return preg_replace('/\+00:00$/', '', $dateTime->format('c'));
```

Comment:
Would it not make more sense to do ->format(...:: HTML5_FORMAT) rather than preg_replaceing format(c)?

Reply:
Of course not, because wrong time of format, but maybe having a different `const` here with `"Y-m-d\TH:i:s"`

Reply:
This was taken from https://github.com/symfony/symfony/blob/5129c4cf7e294b1a5ea30d6fec6e89b75396dcd2/src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToRfc3339Transformer.php#L48. I just removed the 'Z' so it becomes a valid html5 datetime-local format. The idea was to remove DateTimeToRfc3339Transformer in a second refactoring step

Reply:
I see that, but seems like it would be more efficient, and obvious what is happening, to pass to `format` the format you actually want, rather than the wrong format, and editing it with regex.